unittests/AST/ASTImporterTest.cpp

```diff
@@ -97,6 +97,11 @@ testImport(const std::string &FromCode, const ArgVector &FromArgs,
   ASTContext &FromCtx = FromAST->getASTContext(),
       &ToCtx = ToAST->getASTContext();
 
+  FromAST->getDiagnostics().getClient()->BeginSourceFile(
+      FromCtx.getLangOpts(),FromAST->getPreprocessorPtr().get());
+  ToAST->getDiagnostics().getClient()->BeginSourceFile(
```

Comment:
Is this the change which avoids the assert when we have an "unexptected" diagnostic?